app/controllers/concerns/address_handling.rb

```diff
@@ -3,9 +3,16 @@ module AddressHandling
 
 private
 
+  def filter_addresses
```

Comment:
Feels like it would be worth having a spec/specs to explicitly test the address filtering but I don't think I can see any? Possibly something to add to the address_selections_controller_request_spec?

Reply:
@kmahern I've added another spec in the second last commit